api/envoy/extensions/regex_engine/v3/google_re2.proto

```diff
@@ -0,0 +1,34 @@
+syntax = "proto3";
+
+package envoy.extensions.regex_engine.v3;
+
+import "google/protobuf/wrappers.proto";
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.regex_engine.v3";
+option java_outer_classname = "GoogleRe2Proto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/regex_engine/v3;regex_enginev3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Google RE2]
+
+// Google's `RE2 <https://github.com/google/re2>`_ regex engine. The regex string must adhere to
+// the documented `syntax <https://github.com/google/re2/wiki/Syntax>`_. The engine is designed
+// to complete execution in linear time as well as limit the amount of memory used.
+//
+// Envoy emits two stats for tracking the program size of regexes: the histogram `re2.program_size`,
+// which records the program size, and the counter `re2.exceeded_warn_level`, which is incremented
+// each time the program size exceeds the warn level threshold.
+message GoogleRE2 {
+  // This field controls the RE2 "program size" which is a rough estimate of how complex a
+  // compiled regex is to evaluate. A regex that has a program size greater than the configured
+  // value will fail to compile. In this case, the configured max program size can be increased
+  // or the regex can be simplified. If not specified, the default is 100.
+  google.protobuf.UInt32Value max_program_size = 1;
```

Comment:
We already deprecated `max_program_size` in GoogleRE2 and states that should be performed on the management server. So no need to undeprecate these two field?

Reply:
In fact, I add an new `GoogleRE2` message (api/envoy/extensions/regex_engine/v3/google_re2.proto), since I think it should be located outside `RegexMatcher`. And it is also a compatibility concern of xDS API, since we must change both Envoy's API and xDS's API to modify the `GoogleRE2` in `RegexMatcher`.

Reply:
I'm not asking you to modify `GoogleRE2` in `RegexMatcher`, since those fields are already deprecated, so you don't need them in this new `GoogleRE2` message.

Reply:
In fact, we deprecate these fields but we still bring them back in #11187 as runtime values. Now we have decided to deprecated these runtime values in #20168, I think it will be proper to set them in a bootstrap option. https://github.com/envoyproxy/envoy/blob/c9c46bb41130eff0c84dcde0d609dd5c145ac34a/source/common/runtime/runtime_features.cc#L97-L100

Reply:
We are not intended to bring them back, it's there and deprecated just for compatibility. The document here clearly states that's not Envoy's responsibility but control-planes'. So we should just add them in a new message.

cc @alyssawilk

Reply:
Done.